src/main/java/org/myrobotlab/net/CommunicationManager.java

```diff
@@ -67,7 +67,7 @@ final public void send(final Message msg) {
 
     ServiceInterface sw = Runtime.getService(msg.getName());
     if (sw == null) {
-      log.error("could not find service {} to process {} from sender {} - tearing down route", msg.name, msg.method, msg.sender);
+      log.info("could not find service {} to process {} from sender {} - tearing down route", msg.name, msg.method, msg.sender);

```

Comment:
This probably is a warning level ..  someone is trying to send a message to an unknown service, right?  that indicates a bug somewhere.

Reply:
when a service is released, it currently does not scan through all services, and all subscriptions to see if there are any hanging references.
When a message is routed to a service which does not exist, the error will cause the msg sendor to tear down the route ..

Having services removed or added is potentially common, and the framework should allow graceful (or graceful as possible) releasing of services.

Msg routing can be handled at a higher level, for example, if you depend on a service and get a onReleased msg from the framework, the service can do what it thinks is appropriate..